lib/iris/analysis/cartography.py

```diff
@@ -764,6 +764,7 @@ def project(cube, target_proj, nx=None, ny=None):
 
     # Copy metadata across
     new_cube.metadata = cube.metadata
+    new_cube.fill_value = cube.fill_value
```

Comment:
Again, `project` is not lazy, so I reasoned that `dtype` doesn't matter here ...